components/eventing-controller/controllers/backend/reconciler_internal_unit_test.go

```diff
@@ -337,6 +336,209 @@ func Test_updateMutatingValidatingWebhookWithCABundle(t *testing.T) {
+			givenSecrets:     nil,
+			wantError:        true,
+		},
+		// secret is found but some of the required data is missing
+		{
+			name:             "eventing auth manager enabled, and secret exists with missing data",
+			givenFlagEnabled: true,
+			givenSecrets: []*corev1.Secret{
+				// required secret
+				{
+					ObjectMeta: metav1.ObjectMeta{
+						Name:      defaultEventingWebhookAuthSecretName,
+						Namespace: defaultEventingWebhookAuthSecretNamespace,
+					},
+					Data: map[string][]byte{
+						secretKeyClientID: []byte("test-client-id-0"),
+						// missing data
+					},
+				},
+				// not required secret
+				{
+					ObjectMeta: metav1.ObjectMeta{
+						Name:      getOAuth2ClientSecretName(),
+						Namespace: deployment.ControllerNamespace,
+					},
+					Data: map[string][]byte{
+						secretKeyClientID:     []byte("test-client-id-1"),
+						secretKeyClientSecret: []byte("test-client-secret-1"),
+						secretKeyTokenURL:     []byte("test-token-url-1"),
+					},
+				},
+			},
+			wantError: true,
+		},
+		{
+			name:             "eventing auth manager disabled, and secret exists with missing data",
+			givenFlagEnabled: false,
+			givenSecrets: []*corev1.Secret{
+				// not required secret
+				{
+					ObjectMeta: metav1.ObjectMeta{
+						Name:      defaultEventingWebhookAuthSecretName,
+						Namespace: defaultEventingWebhookAuthSecretNamespace,
+					},
+					Data: map[string][]byte{
+						secretKeyClientID:     []byte("test-client-id-0"),
+						secretKeyClientSecret: []byte("test-client-secret-0"),
+						secretKeyTokenURL:     []byte("test-token-url-0"),
+					},
+				},
+				// required secret
+				{
+					ObjectMeta: metav1.ObjectMeta{
+						Name:      getOAuth2ClientSecretName(),
+						Namespace: deployment.ControllerNamespace,
+					},
+					Data: map[string][]byte{
+						secretKeyClientID: []byte("test-client-id-1"),
+						// missing data
+					},
+				},
+			},
+			wantError: true,
+		},
+		// secret is found with all the required data
+		{
+			name:             "eventing auth manager enabled, and secret exists with all data",
+			givenFlagEnabled: true,
+			givenSecrets: []*corev1.Secret{
+				// required secret
+				{
+					ObjectMeta: metav1.ObjectMeta{
+						Name:      defaultEventingWebhookAuthSecretName,
+						Namespace: defaultEventingWebhookAuthSecretNamespace,
+					},
+					Data: map[string][]byte{
+						secretKeyClientID:     []byte("test-client-id-0"),
+						secretKeyClientSecret: []byte("test-client-secret-0"),
+						secretKeyTokenURL:     []byte("test-token-url-0"),
+					},
+				},
+				// not required secret
+				{
+					ObjectMeta: metav1.ObjectMeta{
+						Name:      getOAuth2ClientSecretName(),
+						Namespace: deployment.ControllerNamespace,
+					},
+					Data: map[string][]byte{
+						secretKeyClientID:     []byte("test-client-id-1"),
+						secretKeyClientSecret: []byte("test-client-secret-1"),
+						secretKeyTokenURL:     []byte("test-token-url-1"),
```

Comment:
nitpick: maybe delete token url as oauth2 secret doesn't have it normally

Reply:
As agreed, we will leave it like that.